### DISCUSSION TITLE
Experimental OSProcessStreamConnectionProvider

### DISCUSSION THREAD

**Discussion by angelozerr** [redhat-developer/lsp4ij#541] [Announcements]
Hi all,

I have add a new way to connect to a language server with stdio with [OSProcessStreamConnectionProvider](
https://github.com/redhat-developer/lsp4ij/blob/main/docs/DeveloperGuide.md#osprocessstreamconnectionprovider) which will be available in 0.6.0 (it should be released next week).

LSP4IJ uses this connection provider instead of existing [ProcessStreamConnectionProvider](
https://github.com/redhat-developer/lsp4ij/blob/main/docs/DeveloperGuide.md#processstreamconnectionprovider).

IMHO I think it is better to use it, because it uses OSProcessHandler from IntelliJ which is used in Run configuration for instance. 

Please try it and give us feedback.

cc @CppCXY @InSyncWithFoo @seachicken @KUGDev @AlexWeinstein92 @aparnamichael @MituuZ @turkeylurkey @mrglavas @koxudaxi @FalsePattern @onriv @kristofvb

**Comment by mrglavas** [comment]
@angelozerr We're about to have a release of Liberty Tools which we have tested with LSP4iJ 0.5.0. All of our language server classes currently extend `ProcessStreamConnectionProvider`. Can you confirm that the introduction of this new API does not have an effect on the existing connection providers? Will `ProcessStreamConnectionProvider` still be supported by LSP4iJ 0.6.0?

**Reply by angelozerr** [reply]
Don't worry about that, ProcessStreamConnectionProvider will be never removed (see doc at https://github.com/redhat-developer/lsp4ij/blob/main/docs/DeveloperGuide.md#processstreamconnectionprovider).

I wanted just to communicate to the community that it exists a new implementation of ProcessStreamConnectionProvider.

**Reply by mrglavas** [reply]
Thanks for clarifying.

**Comment by FalsePattern** [comment]
I just tested the new connection provider, it seems to be working fine with zls too. I will update ZigBrains to use it once 0.6.0 is released.

**Reply by angelozerr** [reply]
Thanks for your feedback. 0.6.0 has been released, we need just wait for aprrove from JetBrains.

**Reply by FalsePattern** [reply]
I've released ZB 17.3.0 built against LSP4IJ 0.6.0, both `OSProcessStreamConnectionProvider` and document symbols seem to be working properly with Zig+ZLS

**Reply by angelozerr** [reply]
Nice! Thanks for your feedback.
